sphinx/writers/latex.py

```diff
@@ -462,12 +468,20 @@ def declare_package(packagename, options=None):
             if tocdepth >= SECNUMDEPTH:
                 # Increase secnumdepth if tocdepth is depther than default SECNUMDEPTH
                 self.elements['secnumdepth'] = '\\setcounter{secnumdepth}{%d}' % tocdepth
-        if getattr(document.settings, 'contentsname', None):
-            self.elements['contentsname'] = \
-                self.babel_renewcommand('\\contentsname', document.settings.contentsname)
         # allow the user to override them all
         self.check_latex_elements()
         self.elements.update(builder.config.latex_elements)
```

Comment:
Sorry for late response. I reviewed this now.
How about move this line after initialization of `self.elements`?
Then the trick like `3.1415` is not needed.

There are only two keys which are overrided by writer before this line; `fncychap` and `fontpkg`.
So it will go better to prevent overriding.
```
        if builder.config.language and 'fncychap' not in builder.config.latex_elements:
            # use Sonny style if any language specified
            self.elements['fncychap'] = '\\usepackage[Sonny]{fncychap}'
```
```
                # Times fonts don't work with Cyrillic languages
                if self.babel.uses_cyrillic() and 'fontpkg' not in builder.config.latex_elements:
                    self.elements['fontpkg'] = ''
```


Until `multilingual` introduced, it was reasonable to apply user settings after setting babel.
Because setting babel does not depends on user settings.
But, from 1.5, setting babel depends on user settings. So I think moving the process to top is simple and understandable.